File: advent2019/01/puzzle.py
def parse(lines):
    ans = []
    for line in lines:
        ans.append(int(line))
    return ans
def solve(data):
    total = 0
    for num in data:
        total = total + num
    return total
def solve_p2(data):
    s = set()
    freq = 0
    s.add(freq)
    while True:
        for change in data:
            freq += change
            if freq in s:
                return freq
            s.add(freq)

File: advent2019/01/puzzle.js
const solve = (masses) => {
  return masses.reduce((sum, mass) => {
    const fuel = Math.floor(mass / 3) - 2;
    return sum + fuel;
  }, 0);
}
const parse = (lines) => {
  return lines.map((line) => parseInt(line)).filter((num) => num === num);
};
const solve_p2 = (masses) => {
  return masses.reduce((sum, mass) => {
    let fuelSum = 0;
    let fuel = Math.floor(mass / 3) - 2;
    while (fuel >= 0) {
      fuelSum += fuel;
      fuel = Math.floor(fuel / 3) - 2;
    }
    return sum + fuelSum;
  }, 0);
}
module.exports = {solve, parse, solve_p2};
